### ISSUE TITLE
Trackpad is incorrectly listed as a pointer device and is not picked up by synaptics or mtrack. 

### ISSUE THREAD

**Issue by Pizza-Boy** [voidlinux/void-packages#3897]
Like the title says, my trackpad on my mid-2012 macbook pro appears to be identified as a pointer instead of a touchpad and as a result is picked up by the evdev driver instead of synaptics or mtrack. 

My Xorg log says "Apple Inc. Apple Internal Keyboard/Trackpad: applying InputClass " evdev pointer catchall""

I tried commenting out everything except the lines relevant to the keyboard in 10-evdev.conf (which is placed in the /etc/X11/xord.conf.d directory), which didn't fix the problem. 

Also not an issue I've had with any other distros on this machine. 

It sounds a bit like the problem listed here: https://github.com/voidlinux/void-packages/issues/923 but that problem should've been patched out of the kernel a while ago.

**Comment by Vaelatern** [comment]
Have you tried specifying an input class by any chance? I'm not sure how, I generally avoid Xorg configuration.

**Comment by Pizza-Boy** [comment]
Yeah, guess I should've specified first that I have a basic 10-mtrack.conf set up that should be setting an input class correctly--the same setup works on Arch without any problems.

**Comment by Pizza-Boy** [comment]
Alright, I seem to have figured out the issue (I think). For whatever reason the initrd generated by default wasn't correctly loading my trackpad and keyboard and that was causing driver issues when they were being loaded by Xorg. I solved this by installing mkinitcpio-udev and rebuilding the initrd.

Considering that this is a relatively easy to solve issue that isn't very intuititive would it be helpful if I wrote a little wiki entry on how to solve it?

**Comment by Vaelatern** [comment]
Is there no dracut module that could load this? I'm surprised you had to go to mkinitcpio.

**Comment by Duncaen** [comment]
We use `dracut` to generate the initram by default and it includes udev rules if you don't configure it otherwise.
That can be verified with the `lsinitrd` command.

**Comment by Pizza-Boy** [comment]
Alright, so generating an image via `dracut`with the defaults generates an image that gives me the errors I mentioned above. Using the command 'mkinitcpio -g initramfs-4.4.6_1.img' resolves those errors.

I can try and print the output of the messages from running dracut if that'd help.

**Comment by Vaelatern** [comment]
Can you print the output of running `lsinitrd` on the dracut generated initramfs?

**Comment by Pizza-Boy** [comment]
Sure. Here's the output that `lsinitrd` gives me:

Image: /boot/initramfs-4.4.6_1.img: 32M
========================================================================
/usr/bin/lsinitrd: line 156: /boot/initramfs-4.4.6_1.img: Permission denied
/usr/bin/lsinitrd: line 182: /boot/initramfs-4.4.6_1.img: Permission denied
xzcat: /boot/initramfs-4.4.6_1.img: Permission denied
Version: 

Arguments: xzcat: /boot/initramfs-4.4.6_1.img: Permission denied

dracut modules:
xzcat: /boot/initramfs-4.4.6_1.img: Permission denied
========================================================================
xzcat: /boot/initramfs-4.4.6_1.img: Permission denied
cpio: premature end of archive
========================================================================

**Comment by Vaelatern** [comment]
... Do you have a highly customized dracut? And perhaps you can check your file, and notice you should have run with `sudo` so you don't get permission denied.

**Comment by Pizza-Boy** [comment]
That output came from running the command `sudo dracut`. I haven't done anything else to customize dracut.

**Comment by xa0** [comment]
Hey,

I have the exact same setup and problem as OP. Haven't been able to try it with mkinitcpio yet as it freezes everything forcing me to hard reboot.

Anyway here's output of lsinitrd:
https://gist.github.com/xa0/b80acb04db5c8a0a1dbb

The only change I did to dracut is add hostonly=yes in /etc/dracut.conf.d/hostonly as the installation guide says.

**Comment by bqv** [comment]
hi xa0, this is entirely unrelated, could you message me?

**Comment by xa0** [comment]
Ok, I got mkinitcpio working now. However, I have to use the `initramfs-4.5.5_1.img.old-dkms` image, the other one freezes Xorg for me.

The trackpad is still not working though, I believe it's because of these kernel messages:
```
[  103.768651] udevd[677]: Unable to EVIOCGABS device "/dev/input/event11"
[  103.768673] udevd[677]: Unable to EVIOCGABS device "/dev/input/event11"
[  103.768686] udevd[677]: Unable to EVIOCGABS device "/dev/input/event11"
[  103.768698] udevd[677]: Unable to EVIOCGABS device "/dev/input/event11"
[  103.776381] udevd[679]: Unable to EVIOCGABS device "/dev/input/event10"
[  103.776397] udevd[679]: Unable to EVIOCGABS device "/dev/input/event10"
[  103.776405] udevd[679]: Unable to EVIOCGABS device "/dev/input/event10"
[  103.776412] udevd[679]: Unable to EVIOCGABS device "/dev/input/event10"
```

And here's the relevant parts of xorg.log:
```
[  1366.147] (II) config/udev: Adding input device Apple Inc. Apple Internal Keyboard / Trackpad (/dev/input/event10)
[  1366.147] (**) Option "Device" "/dev/input/event10"
[  1366.148] (II) input device 'Apple Inc. Apple Internal Keyboard / Trackpad', /dev/input/event10 is tagged by udev as: Mouse
[  1366.148] (II) input device 'Apple Inc. Apple Internal Keyboard / Trackpad', /dev/input/event10 is a pointer caps
[  1366.162] (**) Option "config_info" "udev:/sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.8/2-1.8.3/2-1.8.3:1.2/input/input10/event10"
[  1366.162] (II) input device 'Apple Inc. Apple Internal Keyboard / Trackpad', /dev/input/event10 is tagged by udev as: Mouse
[  1366.163] (II) input device 'Apple Inc. Apple Internal Keyboard / Trackpad', /dev/input/event10 is a pointer caps

[  1366.132] (II) config/udev: Adding input device Apple Inc. Apple Internal Keyboard / Trackpad (/dev/input/event11)
[  1366.132] (**) Option "Device" "/dev/input/event11"
[  1366.133] (II) input device 'Apple Inc. Apple Internal Keyboard / Trackpad', /dev/input/event11 is tagged by udev as: Keyboard
[  1366.133] (II) input device 'Apple Inc. Apple Internal Keyboard / Trackpad', /dev/input/event11 is a keyboard
[  1366.146] (**) Option "config_info" "udev:/sys/devices/pci0000:00/0000:00:1d.0/usb2/2-1/2-1.8/2-1.8.3/2-1.8.3:1.0/0003:05AC:0253.0005/input/input11/event11"
[  1366.147] (II) input device 'Apple Inc. Apple Internal Keyboard / Trackpad', /dev/input/event11 is tagged by udev as: Keyboard
[  1366.147] (II) input device 'Apple Inc. Apple Internal Keyboard / Trackpad', /dev/input/event11 is a keyboard
```

I have also tried manually configuring Xorg to no avail.

@Pizza-Boy Any ideas?

**Comment by iprokin** [comment]
Same problem here.
Regenerating initramfs with mkinitcpio does not solve the issue for me.

**Comment by d-kraus** [comment]
Hello,
i have the same problem. 
The touchpad of my mid-2012 macbook pro is identified as a pointer instead of a touchpad.

Is there any solution at the  moment?
Using mkinitcpio did not work for me.

**Comment by d-kraus** [comment]
I managed to get the touchpad working by including the bcm5974 driver when building the initramfs.
I still get the 'Unable to EVIOCGABS device' error for the keyboard, but it works nevertheless.

**Comment by Pizza-Boy** [comment]
Hi @d-kraus, what steps did you take to include the bcm5974 driver during the initramfs build? And was this using dracut or mkinitcpio? I can try and replicate it to see if that works on my end.

**Comment by d-kraus** [comment]
I just did
`sudo dracut --force --add-drivers bcm5974`

**Comment by Pizza-Boy** [comment]
ah, alright. I'll give it a test today and see if that works on a fresh installation.

**Comment by Pizza-Boy** [comment]
I used a similar method to @d-kraus (although I just added the bcm5974 driver to a conf file for Dracut so it's loaded each time the init file is generated). That appears to fix the issues.

I have run into some other issues with `xf86-input-synaptics` and `xf86-input-mtrack`, namely, that you have to remove the synaptics driver to load the mtrack driver and trying to remove the synaptics driver throws up dep errors with `xorg` (which woudl be resolved by having mtrack installed). And some issues with forcing the removal of synaptics. Should I open a new issue for those?

**Comment by lipidity** [comment]
Same problem for me on Macbook5,1
Fixed with:
`sudo dracut --force --add-drivers bcm5974`
